web/src/ui_init.js

```diff
@@ -280,7 +284,20 @@ export function initialize_kitchen_sink_stuff() {
 
     $(window).on("resize", _.throttle(resize.handler, 50));
 
-    // Scrolling in overlays. input boxes, and other elements that
+    // Stop propagation of the previous "wheel" event handler for popovers
+    // with scrollable content (e.g. emoji picker, stream picker, giphy) and
+    // message edit form preview/textarea. These aren't handled by the event
+    // handler below because they aren't in the DOM when the web-app is
+    // initialized.
+    message_viewport.$scroll_container.on(
+        "wheel",
+        ".simplebar-content, .scrolling_list, textarea",
```

Comment:
Shouldn't we target the events more precisely here to avoid this event being called accidentally? For example, using a selector like `div[data-tippy-root]` or `.popover` would ensure that we only stop propagation for the area inside the popovers.

Reply:
I did consider that, but one trade-off is that the popover close/hide behavior is then different at the top/bottom of the message feed with the "wheel" event vs in the middle of the feed when a "scroll" event is triggered.

<details>
<summary>Popovers with `.popover` and `div[data-tippy-root]` selectors</summary>

**When the mouse icon is stationary/nothing is happening, assume that "wheel" events are being triggered.**

[Screencast from 2023-08-10 12-46-39.webm](https://github.com/zulip/zulip/assets/63245456/7b0875cc-cc6a-4cba-a06c-9ff0a3ace739)
</details>

Maybe we want the event handler for the middle of the feed to be more aware of if the scroll is triggered while the mouse is over a popover as well?

Reply:
I'm not sure I understand your concerns about the `.popover` and `div[data-tippy-root]` selectors. I tested them, and everything seems smooth. Could you please explain the differences you encounter when changing the selectors?

I also want to say that I'm not completely against targeting `".simplebar-content, .scrolling_list, textarea"`. I just wanted to point out that we should be careful with stopping propagation for them. Using more specific selectors will make it safer and help avoid unpredictable behavior elsewhere.

Reply:
Sure. Here's the difference as I see it:

- If my mouse cursor is over a popover in the middle of the feed (aka I'm not near the top or bottom), then when the document scroll is triggered by my using the mouse wheel, the popover immediately closes (if there isn't a scrollbar where I'm hovering with the mouse in the popover).
- If my mouse cursor is over a popover at the top/bottom of the feed, then:
  - with the more specific selectors, the popover doesn't close when I trigger a wheel event by using the mouse wheel. I have to move my mouse cursor off the popover to cause moving the mouse wheel to close the popover.
  - with the `".simplebar-content"` selector, the popover will close in the same way/places as the scroll event in the middle of the feed when I use my mouse wheel. 

Steps to reproduce:
1. With the current PR changes, in the **All messages view**, go to the middle of the feed and open popovers (emoji, usercard, message actions) and use a touchpad/mouse wheel to trigger scroll events while the mouse is visibily over the popovers.
2. In a stream/topic view with one message (e.g. no document scroll events), do the same testing as you did in the **All messages** view.
3. Now change the PR to use the more specific `.popover` selector and repeat the two steps above.

Reply:
Would it make sense for this to be a check on the parents of `e` in the main `message_viewport.$scroll_container` handler above to return early rather than doing `stopPropagation`? I feel a bit uncomfortable with the `stopPropagation` here, because it might prevent other code from processing these events.

FWIW we also now only have tippy popovers -- so if we wanted a check for a popover, `[data-tippy-root]` now works. (Though we still have bootstrap typeahead not using Tippy!).

Reply:
Alternatively, what if we changed the next block from `$(".modal-body, .scrolling_list, input, textarea").on("wheel"` to `$("body"),on("wheel"`, ".modal-body, .scrolling_list, input, textarea") {` -- we use that registration method for most handlers that we want to apply even to elements that might be created in the future.

Reply:
The `$("body").on("wheel"` registration sounds better to me, too.